store/tikv/prewrite.go

```diff
@@ -56,7 +56,9 @@ func (c *twoPhaseCommitter) buildPrewriteRequest(batch batchMutations, txnSize u
 		}
 		isPessimisticLock[i] = m.IsPessimisticLock(i)
 	}
-	var minCommitTS = c.minCommitTS
+	c.mu.Lock()
+	minCommitTS := c.minCommitTS
```

Comment:
The race can happen only in async-commit. Another solution is if it's an async-commit transaction, don't read it and set minCommitTs to 0.

BTW, I find a bug in supporting external consistency. The minCommitTs got from PD is not persisted in TiKV. It can break external consistency in recovery.

Reply:
If it's an async-commit, we still need to read it. This value can be set to ensure external consistency.

Reply:
```go
	if c.isAsyncCommit() {
		if batch.isPrimary {
			req.Secondaries = c.asyncSecondaries()
		}
		req.UseAsyncCommit = true
		// The async commit can not be used for large transactions, and the commit ts can't be pushed.
		req.MinCommitTs = 0
	}
```

Reply:
Oh.... I see.